### PR TITLE
Make AFK/Away commands show your AFK/Away beside your name. 

### PR DESCRIPTION
**Pull request by psnsVGC**

### CODE REVIEW

File: chat-plugins/afk.js [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+
+		newName += ' - ' + status;
+		if (newName.length > 18) return this.sendReply("\"" + target + "\" is too long to use as your away status.");
+
+		// forcerename any possible impersonators
+		let targetUser = Users.getExact(user.userid + target);
+		if (targetUser && targetUser !== user && targetUser.name === user.name + ' - ' + target) {
+			targetUser.resetName();
+			targetUser.send("|nametaken||Your name conflicts with " + user.name + (user.name.substr(-1) === "s" ? "'" : "'s") + " new away status.");
+		}
```
Comment (Slayer95) [inline chat-plugins/afk.js]: This can be easily abused

File: chat-plugins/afk.js [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+	},
+
+	nerding: function (target, room, user) {
+		this.parse('/away NERDING');
+	},
+
+	coding: function (target, room, user) {
+		this.parse('/away CODING');
+	},
+};
```
Comment (Slayer95) [inline chat-plugins/afk.js]: All these commands may actually be implemented as aliases of ``away``, but they seem very unneeded tbh.

File: config/colour.js [hunk 3]
```diff
@@ -0,0 +1,164 @@
+module.exports = hashColor;
+
+function MD5(e) {
```
Comment (Slayer95) [inline config/colour.js]: We don't need to implement MD5 in Node.js

File: chat-plugins/afk.js [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+
+	back: function (target, room, user) {
+		if (!user.isAway) return this.sendReply("You are not set as away.");
+		user.isAway = false;
+
+		let newName = user.name;
+		let statusIdx = newName.search(/\s\-\s[\u24B6-\u24E9\u2460-\u2468\u24EA]+$/);
+		if (statusIdx < 0) {
+			user.isAway = false;
+			if (user.can('lock', null, room)) this.add("");
```
Comment (Slayer95) [inline chat-plugins/afk.js]: What's the point of calling ``this.add``?

**Comment by Slayer95** [comment]
@Morfent , do you want this withdrawn?

**Comment by Morfent** [comment]
no, sorry, my mistake; this is fine under the mit license

**Comment by Morfent** [comment]
i just need to confirm whether or not it can be under the MIT license; otherwise I'm fine with it.

**Comment by Zarel** [comment]
I already warned you that the next time you opened multiple pull requests, I'd ban you from the repository...

**Comment by Zarel** [comment]
@Morfent, the license legality isn't a problem, but there's no reason to credit him if you were the author.

**Comment by Zarel** [comment]
The way I'd probably implement custom away statuses is probably with a `|` separator, and maybe some other stuff... I suppose we could do it like this and then use the client to convert them back to normal... Still, though, I don't think very much of this is usable, so I'm closing this...

**Comment by kotarou3** [comment]
Code is horrible :-1:

**Comment by Zarel** [comment]
Some parts of the code are actually not bad; the transformation to circle-letter Unicode is a pretty neat idea; I assume those are the parts Morfent wrote?

**Comment by Irraquated** [comment]
@Morfent yeah it's your old code you wrote for nimbus a while back.

**Pull request closed without merging**